src/ServicePulse.Host/vue/src/components/failedmessages/AllDeletedMessages.vue

```diff
@@ -127,57 +127,56 @@ function updateMessagesScheduledDeletionDate(messages) {
 }
 
 function numberSelected() {
-  return messageList?.value?.getSelectedMessages()?.length ?? 0;
+  return messageList.value?.getSelectedMessages()?.length ?? 0;
 }
 
 function selectAll() {
-  messageList.value.selectAll();
+  messageList.value?.selectAll();
 }
 
 function deselectAll() {
-  messageList.value.deselectAll();
+  messageList.value?.deselectAll();
 }
 
 function isAnythingSelected() {
-  return messageList?.value?.isAnythingSelected();
+  return messageList.value?.isAnythingSelected();
 }
 
 async function restoreSelectedMessages() {
   changeRefreshInterval(1000);
-  const selectedMessages = messageList.value.getSelectedMessages();
+  const selectedMessages = messageList.value?.getSelectedMessages() ?? [];
   selectedMessages.forEach((m) => (m.restoreInProgress = true));
-  useShowToast("info", "Info", "restoring " + selectedMessages.length + " messages...");
+  useShowToast(TYPE.INFO, "Info", `restoring ${selectedMessages.length} messages...`);
 
   await usePatchToServiceControl(
     "errors/unarchive",
     selectedMessages.map((m) => m.id)
   );
-  messageList.value.deselectAll();
+  messageList.value?.deselectAll();
 }
 
-function periodChanged(period) {
+function periodChanged(period: PeriodOption) {
   selectedPeriod.value = period;
   cookies.set("all_deleted_messages_period", period);
 
   loadMessages();
 }
 
 async function getConfiguration() {
-  const response = await useFetchFromServiceControl("configuration");
-  const data = await response.json();
+  const [, data] = await useTypedFetchFromServiceControl<Configuration>("configuration");
   configuration.value = data;
 }
 
 function isRestoreInProgress() {
   return messages.value.some((message) => message.restoreInProgress);
 }
 
-function changeRefreshInterval(milliseconds) {
-  if (typeof refreshInterval !== "undefined") {
-    clearInterval(refreshInterval);
+function changeRefreshInterval(milliseconds: number) {
+  if (refreshInterval != null) {
```

Comment:
it seems `refreshInterval` is defined as `number | undefined`

Reply:
yep, this check is for null or undefined